bids-validator/tests/json.spec.js

```diff
@@ -467,6 +467,83 @@ describe('JSON', function() {
     relativePath: '/dataset_description.json',
   }
 
+  it('dataset_description.json should validate DatasetLinks', function() {
+    var jsonObj = {
+      Name: 'Example Name',
+      BIDSVersion: '1.4.0',
+      DatasetLinks: {
+        mylink: 'https://www.google.com',
+        deriv1: 'derivatives/derivative1',
+        phantoms: 'file:///data/phantoms',
+        ds000001: 'doi:10.18112/openneuro.ds000001.v1.0.0',
+      },
+    }
+    jsonDict[dataset_description_file.relativePath] = jsonObj
+    validate.JSON(dataset_description_file, jsonDict, function(issues) {
+      assert(issues.length === 0)
+    })
+  })
+
+  it('dataset_description.json should raise on empty key in DatasetLinks', function() {
+    var jsonObj = {
+      Name: 'Example Name',
+      BIDSVersion: '1.4.0',
+      DatasetLinks: {
+        mylink: 'https://www.google.com',
+        '': 'https://www.yahoo.com',
+      },
+    }
+    jsonDict[dataset_description_file.relativePath] = jsonObj
+    validate.JSON(dataset_description_file, jsonDict, function(issues) {
+      assert(issues.length === 2)
+      assert(
+        issues[0].evidence ==
+          '.DatasetLinks should NOT be shorter than 1 characters',
+      )
+      assert(issues[1].evidence == ".DatasetLinks property name '' is invalid")
+    })
+  })
+
+  it('dataset_description.json should raise on wrong key in DatasetLinks', function() {
+    var jsonObj = {
+      Name: 'Example Name',
+      BIDSVersion: '1.4.0',
+      DatasetLinks: 'https://www.google.com',
+    }
+    jsonDict[dataset_description_file.relativePath] = jsonObj
+    validate.JSON(dataset_description_file, jsonDict, function(issues) {
+      assert(issues.length === 1)
+      assert(issues[0].evidence == '.DatasetLinks should be object')
+    })
+  })
+
+  it('dataset_description.json should raise on invalid values in DatasetLinks', function() {
+    var jsonObj = {
+      Name: 'Example Name',
+      BIDSVersion: '1.4.0',
+      DatasetLinks: {
+        mylink1: 'https://www.google.com',
+        mylink2: 1,
+        mylink3: 'nope',
```

Comment:
I guess this is now valid. Can either drop it or make it invalid.

```suggestion
        mylink3: ':/nope',
```

Reply:
I'll drop it and change it to a test that fails for keys like `I am a key with whitespace and not alphanumeric "§%`